template-compose/.github/workflows/review_pull_request.yml

```
@@ -0,0 +1,76 @@
+name: Review pull request
+
+on:
+  pull_request:
+    types: [ opened, reopened, ready_for_review, synchronize ]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  review_pull_request:
+    if: github.event.pull_request.draft == false
+    name: Review pull request
+    runs-on: ubuntu-latest
+    environment: dev
+    timeout-minutes: 30
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
+
+      - name: Cache Gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches/modules-*
+            ~/.gradle/caches/jars-*
+            ~/.gradle/caches/build-cache-*
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+
+      - name: Run Detekt
+        run: ./gradlew detekt
+
+      - name: Run Android Lint
+        run: ./gradlew lintDevDebug
+
+      - name: Run unit tests with Kover
+        run: ./gradlew koverXmlReport
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '2.7'
```

We can cache with fewer steps if you want.
```suggestion
        with:
          ruby-version: '2.7'
          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
```